lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -56,6 +60,80 @@ LogicalResult LogicEquivalenceCheckingOp::verifyRegions() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ClockedAssertLikeOps
+//===----------------------------------------------------------------------===//
+
+namespace {
+// Verify function for clocked assert / assume / cover ops.
+// This checks that they do not contiain any nested clocks or disable operations
+// Clocked assertlike ops are a simple form of assertions that only
+// contain one clock and one disable condition.
+struct ClockedAssertLikeOp {
+  static LogicalResult verify(Operation *clockedAssertLikeOp) {
+    // Used to perform a DFS search through the module to visit all operands
+    // before they are used
+    llvm::SmallMapVector<Operation *, OperandRange::iterator, 16> worklist;
+
+    // Keeps track of operations that have been visited
+    llvm::DenseSet<Operation *> handledOps;
+
+    Operation *property = clockedAssertLikeOp->getOperand(0).getDefiningOp();
+
+    // Fill in our worklist
+    worklist.insert({property, property->operand_begin()});
+
+    // Process the elements in our worklist
+    while (!worklist.empty()) {
```

Comment:
Verifier shouldn't follow use-def chain(please see https://mlir.llvm.org/getting_started/DeveloperGuide/#ir-verifier). It will eventually blow up compilation time. I don't have a good alternative solution other than separating the verification into a pass. I think we can make the verification local if types contain clock domain information but probably it's not worth doing that.